byway/src/main/java/com/google/sps/data/Trip.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+
+/**
+ * A class to make a Trip type, containing
+ * a specific id, destinations, interests and routes 
+ * to be used when updating a trip.
+ */
+public final class Trip {
+
+  private String id;
+  private String start;
+  private ArrayList<String> destinations;
+  private ArrayList<String> interests;
+  private ArrayList<String> route;
+
+  public Trip(String id, String start, ArrayList<String> destinations,
+              ArrayList<String> interests, ArrayList<String> route) {
+    this.id = id;
+    this.start = start;
```

Comment:
Static import `com.google.common.base.Preconditions.checkNotNull` and wrap these assignments as appropriate:

`this.start = checkNotNull(start, "start");` (note the `"start"` is the error message that will appear if `start` is null. We usually just write the name of the parameter that was null when doing these kind of checks, since that's sufficient to figure out which thing caused the problem.)

Reply:
Fixed